ethnobiology-and-conservation.csl

```diff
@@ -1,116 +1,133 @@
+          <text variable="publisher-place" suffix="."/>
+        </else-if>
+        
+        <else-if type="chapter">
+          <text macro="editor" prefix="In: "/>
+          <text variable="container-title" font-style="italic" suffix=". "/>
+          <text variable="edition" suffix=" ed. " form="ordinal"/>
+          <text variable="publisher" suffix=", "/>
+          <text variable="publisher-place" suffix=". "/>
+          <label variable="page" form="short" suffix=" "/>
+          <text variable="page"/>
+        </else-if>
+        
+        <else-if type="thesis">
+          <text variable="genre" suffix=". "/>
           <text variable="publisher" suffix=", "/>
-          <text variable="publisher-place" suffix=" "/>
+          <text variable="publisher-place" suffix="."/>
         </else-if>
+        
         <else-if type="paper-conference">
           <text variable="event" prefix=" " suffix=" "/>
           <text variable="volume" suffix=":"/>
           <text variable="page"/>
         </else-if>
-        <else-if type="chapter">
-          <text macro="editor"/>
-          <text variable="container-title" prefix=" " suffix=". "/>
-          <text variable="edition" suffix=" ed. "/>
-          <text variable="publisher" suffix=", "/>
-          <text variable="publisher-place" suffix=", "/>
-          <label strip-periods="false" suffix=" " variable="page" form="symbol"/>
-          <text variable="page" suffix=" "/>
-        </else-if>
+        
         <else-if type="webpage">
           <text variable="URL" prefix="[" suffix="] "/>
-          <date form="text" variable="accessed" prefix="Accessed "/>
-        </else-if>
-        <else-if type="article article-journal article-magazine article-newspaper" match="any">
-          <text variable="container-title" font-style="italic" suffix=" "/>
-          <text variable="volume" suffix=":"/>
-          <text variable="page"/>
+          <text term="accessed" text-case="capitalize-first"/>
+          <date variable="accessed">
+            <date-part name="day" suffix=" "/>
+            <date-part name="month" suffix=" "/>
+            <date-part name="year"/>
+          </date>
         </else-if>
-        <else-if type="report" match="any">
+        
+        <else-if type="report">
           <text variable="publisher" suffix=", "/>
           <text variable="publisher-place" suffix=". "/>
           <text variable="URL" prefix="[" suffix="]. "/>
-          <text variable="DOI" prefix="doi: "/>
         </else-if>
+        
         <else>
-          <group suffix=" ">
-            <text variable="container-title" suffix=" " form="short"/>
-            <text variable="volume" suffix=":"/>
-            <text variable="page"/>
-          </group>
+          <text variable="container-title" font-style="italic" suffix=" "/>
+          <text variable="volume" suffix=":"/>
+          <text variable="page"/>
         </else>
       </choose>
+      
+      <if variable="DOI">
```

Comment:
This conditional is not necessary as it would only ever print out the DOI if the condition was also true.